File: APIDemoJavaApp/app/src/main/java/com/leoleo/apidemojavaapp/MyReceiver.java
package com.leoleo.apidemojavaapp;

import android.content.BroadcastReceiver;
import android.content.Context;
import android.content.Intent;
import android.os.LocaleList;
import android.util.Log;

import androidx.core.text.util.LocalePreferences;

public class MyReceiver extends BroadcastReceiver {

    @Override
    public void onReceive(Context context, Intent intent) {
        final String packageName = intent.getStringExtra(Intent.EXTRA_PACKAGE_NAME);
        final LocaleList localeList = intent.getParcelableExtra(Intent.EXTRA_LOCALE_LIST, LocaleList.class);
        final StringBuilder message = new StringBuilder();
        message.append("onReceive: packageName= ").append(packageName).append("\n");
        message.append("localeList = ").append(localeList).append("\n");
        message.append("getHourCycle = ").append(LocalePreferences.getHourCycle()).append("\n");
        message.append("getCalendarType = ").append(LocalePreferences.getCalendarType()).append("\n");
        message.append("getTemperatureUnit = ").append(LocalePreferences.getTemperatureUnit()).append("\n");
        message.append("getFirstDayOfWeek = ").append(LocalePreferences.getFirstDayOfWeek()).append("\n");
        Log.d("TAG", message.toString());
    }
}

File: APIDemoKotlinApp/app/src/main/java/com/leoleo/apidemokotlinapp/MyReceiver.kt
package com.leoleo.apidemokotlinapp

import android.content.BroadcastReceiver
import android.content.Context
import android.content.Intent
import android.os.LocaleList
import android.util.Log
import androidx.core.text.util.LocalePreferences

class MyReceiver : BroadcastReceiver() {
    override fun onReceive(context: Context, intent: Intent) {
        val packageName = intent.getStringExtra(Intent.EXTRA_PACKAGE_NAME)
        val localeList = intent.getParcelableExtra(Intent.EXTRA_LOCALE_LIST, LocaleList::class.java)
        val message = """
            onReceive: packageName= $packageName
            localeList= $localeList
            getHourCycle= ${LocalePreferences.getHourCycle()}
            getCalendarType= ${LocalePreferences.getCalendarType()}
            getCalendarType= ${LocalePreferences.getCalendarType()}
            getTemperatureUnit= ${LocalePreferences.getTemperatureUnit()}
            getFirstDayOfWeek= ${LocalePreferences.getFirstDayOfWeek()}
        """.trimIndent()

        Log.d("TAG", message)
    }
}